pkg/plugin/test-keeper/plugin/event_handler_gh_test.go

```diff
@@ -213,6 +213,36 @@ var _ = Describe("Test Keeper Plugin features", func() {
 			Ω(err).ShouldNot(HaveOccurred())
 		})
 
+		It("should block newly created pull request when deletions are the only changes in test file.", func() {
+			// given
+			gock.New("https://api.github.com").
+				Get("/repos/" + repositoryName + "/pulls/1/files").
+				Reply(200).
+				Body(FromFile("test_fixtures/github_calls/prs/without_tests/deletions_only_changes_in_tests.json"))
+			gock.New("https://api.github.com").
+				Get("/repos/" + repositoryName + "/issues/1/comments").
+				Reply(200).
+				BodyString("[]")
+
+			gock.New("https://api.github.com").
+				Post("/repos/" + repositoryName + "/statuses").
+				SetMatcher(ExpectPayload(toBe(github.StatusFailure, keeper.NoTestsMessage, expectedContext, keeper.NoTestsDetailsLink))).
+				Reply(201) // This way we implicitly verify that call happened after `HandleEvent` call
+
+			gock.New("https://api.github.com").
+				Post("/repos/" + repositoryName + "/issues/1/comments").
+				SetMatcher(ExpectPayload(toHaveBodyWithWholePluginsComment)).
+				Reply(201)
+
+			statusPayload := LoadFromFile("test_fixtures/github_calls/prs/without_tests/status_opened.json")
+
+			// when
+			err := handler.HandleEvent(log, github.PullRequest, statusPayload)
+
+			// then - implicit verification of /statuses call occurrence with proper payload
+			Ω(err).ShouldNot(HaveOccurred())
+		})
+
```

Comment:
I would suggest another test case:

```golang
It("should block newly created pull request when there are changes in the business logic but only deletions in the tests", func() {
			// given
			gock.New("https://api.github.com").
				Get("/repos/" + repositoryName + "/pulls/1/files").
				Reply(200).
				Body(FromFile("test_fixtures/github_calls/prs/without_tests/prod_code_changes_with_deletion_only_in_tests.json"))
			gock.New("https://api.github.com").
				Get("/repos/" + repositoryName + "/issues/1/comments").
				Reply(200).
				BodyString("[]")

			gock.New("https://api.github.com").
				Post("/repos/" + repositoryName + "/statuses").
				SetMatcher(ExpectPayload(toBe(github.StatusFailure, keeper.NoTestsMessage, expectedContext, keeper.NoTestsDetailsLink))).
				Reply(201) // This way we implicitly verify that call happened after `HandleEvent` call

			gock.New("https://api.github.com").
				Post("/repos/" + repositoryName + "/issues/1/comments").
				SetMatcher(ExpectPayload(toHaveBodyWithWholePluginsComment)).
				Reply(201)

			statusPayload := LoadFromFile("test_fixtures/github_calls/prs/without_tests/status_opened.json")

			// when
			err := handler.HandleEvent(log, github.PullRequest, statusPayload)

			// then - implicit verification of /statuses call occurrence with proper payload
			Ω(err).ShouldNot(HaveOccurred())
		})

```

with the `prod_code_changes_with_deletion_only_in_tests.json` file : 

```json
[
  {
    "sha": "56c7705759b68d3c79ebed0bc5cb2ba54ffee6d7",
    "filename": "src/main/java/io/openshift/booster/ApplicationConfig.java",
    "status": "modified",
    "additions": 1,
    "deletions": 23,
    "changes": 24,
    "blob_url": "https://github.com/bartoszmajsak/wfswarm-booster-pipeline-test/blob/0beab1ffd081660b0892764d2cdfcdfb0e9ea8e3/src/main/java/io/openshift/booster/ApplicationConfig.java",
    "raw_url": "https://github.com/bartoszmajsak/wfswarm-booster-pipeline-test/raw/0beab1ffd081660b0892764d2cdfcdfb0e9ea8e3/src/main/java/io/openshift/booster/ApplicationConfig.java",
    "contents_url": "https://api.github.com/repos/bartoszmajsak/wfswarm-booster-pipeline-test/contents/src/main/java/io/openshift/booster/ApplicationConfig.java?ref=0beab1ffd081660b0892764d2cdfcdfb0e9ea8e3",
    "patch": "@@ -1,31 +1,9 @@\n-/*\n- *\n- *  Copyright 2016-2017 Red Hat, Inc, and individual contributors.\n- *\n- *  Licensed under the Apache License, Version 2.0 (the \"License\");\n- *  you may not use this file except in compliance with the License.\n- *  You may obtain a copy of the License at\n- *\n- *  http://www.apache.org/licenses/LICENSE-2.0\n- *\n- *  Unless required by applicable law or agreed to in writing, software\n- *  distributed under the License is distributed on an \"AS IS\" BASIS,\n- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n- *  See the License for the specific language governing permissions and\n- *  limitations under the License.\n- *\n- */\n-\n package io.openshift.booster;\n \n import javax.ws.rs.ApplicationPath;\n import javax.ws.rs.core.Application;\n \n-/**\n- * Resource entry point\n- */\n+\n @ApplicationPath(\"/api\")\n public class ApplicationConfig extends Application {\n-\n }\n-"
  },
  {
    "sha": "eaed6f6c17eca636fd2ac190001495248f053f39",
    "filename": "src/test/java/io/openshift/booster/GreetingServiceTest.java",
    "status": "modified",
    "additions": 0,
    "deletions": 13,
    "changes": 13,
    "blob_url": "https://github.com/bartoszmajsak/wfswarm-booster-pipeline-test/blob/0beab1ffd081660b0892764d2cdfcdfb0e9ea8e3/src/test/java/io/openshift/booster/GreetingServiceTest.java",
    "raw_url": "https://github.com/bartoszmajsak/wfswarm-booster-pipeline-test/raw/0beab1ffd081660b0892764d2cdfcdfb0e9ea8e3/src/test/java/io/openshift/booster/GreetingServiceTest.java",
    "contents_url": "https://api.github.com/repos/bartoszmajsak/wfswarm-booster-pipeline-test/contents/src/test/java/io/openshift/booster/GreetingServiceTest.java?ref=0beab1ffd081660b0892764d2cdfcdfb0e9ea8e3",
    "patch": "@@ -50,17 +50,4 @@ public void test_service_invocation() {\n         Assert.assertTrue(response.readEntity(String.class).contains(\"Hello, World!\"));\n     }\n \n-    @Test\n-    @RunAsClient\n-    public void test_service_invocation_withParam() {\n-        Client client = ClientBuilder.newClient();\n-        WebTarget target = client.target(\"http://localhost:8080\")\n-                .path(\"api\").path(\"greeting\")\n-                .queryParam(\"name\", \"Peter\");\n-\n-        Response response = target.request(MediaType.APPLICATION_JSON).get();\n-        Assert.assertEquals(200, response.getStatus());\n-        Assert.assertTrue(response.readEntity(String.class).contains(\"Hello, Peter!\"));\n-    }\n-\n }"
  }
]
```